warehouse_ros_mongo/mongo_wrapper_ros.py

```diff
@@ -30,55 +32,61 @@ def print_help_message():
     print_help_message()
     sys.exit()
 
-rospy.init_node('mongodb')
-path_param = rospy.get_param('~database_path' , None)
-if path_param is None:
-    path_param = rospy.get_param('~db_path' , default_db_path)
+rclpy.init()
+node = Node('mongodb', allow_undeclared_parameters=True, automatically_declare_parameters_from_overrides=True)
+
+path_param = node.get_parameter_or('database_path' , Parameter('', Parameter.Type.STRING, 'None'))
+if path_param.value == 'None':
+    path_param = node.get_parameter_or('db_path' , Parameter('', Parameter.Type.STRING, default_db_path))
 else:
     print('Parameter "database_path" is deprecated. Please use "db_path" instead.')
-dbpath = os.path.expanduser(path_param)
-overwrite = rospy.get_param('~overwrite', False)
+dbpath = os.path.expanduser(path_param.value)
+overwrite = node.get_parameter_or('overwrite', False)
 
 if '--repair' in sys.argv:
-    rospy.loginfo("Repairing database")
+    node.get_logger().info("Repairing database")
     lock_file = '{0}/mongod.lock'.format(dbpath)
     if os.path.exists(lock_file):
-        rospy.loginfo("  Removing lock file")
+        node.get_logger().info("  Removing lock file")
         os.remove(lock_file)
     check_call(['mongodb', 'mongod', '--repair', '--dbpath', dbpath])
-    rospy.loginfo("  Successfully repaired.")
+    node.get_logger().info("  Successfully repaired.")
     sys.exit(0)
 
 # The defaults here should match the ones used by each client library
-port = rospy.get_param('warehouse_port', 27017)
-host = rospy.get_param('warehouse_host', 'localhost')
+port = node.get_parameter_or('warehouse_port', Parameter('', Parameter.Type.INTEGER, 27017))
```

Comment:
Shouldn't this use the new port?

Reply:
Changed this to new port number that aligns with the launch file.